### DISCUSSION TITLE
Disscussion: call zoxide inside yazi

### DISCUSSION THREAD

**Discussion by gapplef** [sxyazi/yazi#502] [1. Q&A]
Since the `fd` and `zoxide` in ubuntu's official repository  is  [too old](https://github.com/sxyazi/yazi/discussions/487) to function properly. I am trying to build a `snap` package which bundle all necessary dependencies. After some investigation I managed to do it. The [configuration file](https://github.com/gapplef/yazi/blob/main/snap/snapcraft.yaml) actually quiet simple.

All things work fine except `zoxide`. The problem is `zoxide` is called with interactive selection mode by default in `yazi`, which need a directory database exist. This database is generate automatically once user use `z`/ `zoxide` to change directories.

This process have no problems for `zoxide` user. But for a `yazi` user, who may never used `zoxide` before, there is no way build up or edit the database inside `yazi`. I think some entry to call `zoxide` in normal mode to build up the database or some way to direct edit the database inside `yazi` would be much helpful.

A relate issue is  https://github.com/sxyazi/yazi/issues/310

**Comment by sxyazi** [comment]
I have plans to reimplement zoxide as a Lua plugin and extend its functionality at that time. If all goes well, this will happen in the next target v0.2.1.

**Reply by gapplef** [reply]
Ok, that would be great.

**Reply by ysooqe** [reply]
Just out of curiosity, I have read this in a couple of your responses so far and I am wondering: what is the reasoning behind adding new features through Lua plugins instead of adding them natively?

**Comment by gapplef** [comment]
@sxyazi  Do you have any interests to provide snap support?

I could make a pull request for the `snapcraft.yaml` file and github action to build snap package if welcomed.

**Reply by sxyazi** [reply]
That's great, it would be very convenient for Ubuntu users. Maybe we can add it to Yazi's [release workflow](https://github.com/sxyazi/yazi/blob/main/.github/workflows/release.yml)?

**Reply by gapplef** [reply]
Done https://github.com/sxyazi/yazi/pull/531

**Comment by ysooqe** [comment]
I think there is another way to think about the point OP is making: Regarding zoxide, it covers a very certain use case: having **favorite** directories and jumping to them quickly, but I feel like there is another big use case that is probably not properly covered by zoxide (but people might assume it is): jumping to a directory based on an interactive fuzzy search, without any inherent "bias". Zoxide could be "abused" to do that (e.g. by writing a script that indexes all relevant directories), but it's not optimal, since directories would have to be (re)indexed into zoxide constantly.

I feel like this is what OP is looking for:
I am thinking of something along the lines of the current implementation of "cd --interactive", but with the addition of recursing into all subdirectories and defining an absolute "starting point" (most likely $HOME) so that it is possible to go from anywhere to everywhere really really quickly. It could be done using `fd` to generate a list of directories on the fly and then selecting one through fzf (or any of its cousins like `nucleo`), all of this integrated into the current interface for "cd --interactive".

I'd be happy to work on and submit a PR if this is something that would be accepted.

EDIT: it would be probably even better to create a list of directories on startup and cache it (and update it frequently).

**Reply by lpnh** [reply]
Your description made me think of [Telescope](https://github.com/nvim-telescope/telescope.nvim). I'm not sure if it's directly related, but it could be worth a look. Maybe it could offer some ideas.
